compiler/rustc_pattern_analysis/src/rustc.rs

```diff
@@ -895,6 +895,10 @@ impl<'p, 'tcx: 'p> TypeCx for RustcMatchCheckCtxt<'p, 'tcx> {
             errors::OverlappingRangeEndpoints { overlap: overlaps, range: pat_span },
         );
     }
+
+    fn complexity_exceeded(&self) -> Result<(), Self::Error> {
+        Err(self.tcx.dcx().err("reached pattern complexity limit"))
```

Comment:
Could you make the error point to the `match` that triggered the issue? You can use `self.whole_match_span`

Reply:
Done. :+1: